lib/Echidna/ABI.hs

```diff
@@ -297,16 +294,15 @@ mutateAbiCall = traverse f
 -- | Given a generator taking an @a@ and returning a @b@ and a way to get @b@s associated with some
 -- @a@ from a GenDict, return a generator that takes an @a@ and either synthesizes new @b@s with the
 -- provided generator or uses the 'GenDict' dictionary (when available).
-genWithDict :: (Eq a, Hashable a, MonadReader x m, Has GenDict x, MonadRandom m)
-            => (GenDict -> HashMap a [b]) -> (a -> m b) -> a -> m b
-genWithDict f g t = do
-  let fromDict = uniformMay . M.lookupDefault [] t . f
-  c <- view hasLens
-  fromMaybe <$> g t <*> (bool (pure Nothing) (fromDict c) . (c ^. pSynthA >=) =<< getRandom)
+genWithDict :: (Eq a, Hashable a, MonadRandom m)
+            => GenDict -> HashMap a [b] -> (a -> m b) -> a -> m b
+genWithDict genDict f g t = do
```

Comment:
we should rename the f parameter here because its meaning has changed and is no longer a function